File: shared_resources/src/main/java/com/thewizrd/shared_resources/tasks/TaskUtils.java
package com.thewizrd.shared_resources.tasks;

import com.google.android.gms.tasks.CancellationToken;

public class TaskUtils {
    public static void throwIfCancellationRequested(CancellationToken token) throws InterruptedException {
        if (token != null && token.isCancellationRequested()) {
            throw new InterruptedException();
        }
    }
}

File: shared_resources/src/main/java/com/thewizrd/shared_resources/tasks/TaskUtils.kt
@file:JvmName("TaskUtils")

package com.thewizrd.shared_resources.tasks

import com.google.android.gms.tasks.CancellationToken

@Throws(InterruptedException::class)
fun CancellationToken?.throwIfCancellationRequested() {
    if (this != null && this.isCancellationRequested) {
        throw InterruptedException()
    }
}